types/screens/edit_profile.ts

```diff
@@ -2,18 +2,23 @@
 // See LICENSE.txt for license information.
 
 import type {AvailableScreens} from './navigation';
-import type {FloatingTextInputRef} from '@components/floating_text_input_label';
 import type {FieldProps} from '@screens/edit_profile/components/field';
 import type UserModel from '@typings/database/models/servers/user';
-import type {RefObject} from 'react';
 
-export interface UserInfo extends Record<string, string | undefined | null| boolean> {
+export interface UserInfo {
     email: string;
     firstName: string;
     lastName: string;
     nickname: string;
     position: string;
     username: string;
+    customAttributes: Record<string, CustomAttribute>;
```

Comment:
Might as well make

```
export interface CustomAttributes {
  [key: string]: CustomAttribute;
}
```

Then you can reuse this in a few instances in the user.ts file.

Reply:
added, I also declared the return type which showed a weird return in one of the cases :)